src/dist/dist.rs

```diff
@@ -406,16 +406,13 @@ impl TargetTriple {
 impl std::convert::TryFrom<PartialTargetTriple> for TargetTriple {
     type Error = &'static str;
     fn try_from(value: PartialTargetTriple) -> std::result::Result<Self, Self::Error> {
-        if value.arch.is_some() && value.os.is_some() && value.env.is_some() {
-            Ok(Self(format!(
-                "{}-{}-{}",
-                value.arch.unwrap(),
-                value.os.unwrap(),
-                value.env.unwrap()
-            )))
-        } else {
-            Err("Incomplete / bad target triple")
-        }
+        let arch = value.arch.ok_or("Incomplete / bad target triple")?;
+        let os = value.os.ok_or("Incomplete / bad target triple")?;
+        let triple = match value.env {
```

Comment:
Tiny style nit: I would spell this `Ok(Self(match value.env { .. }))` directly, which avoids a low-value single use binding.

Reply:
Sure, willing to do so.